SCons/Util.py

```diff
@@ -146,23 +146,19 @@ def __getattr__(self, name):
         result = [getattr(x, name) for x in self.data]
         return self.__class__(result)
 
-    def __getitem__(self, index):
-        """
-        This comes for free on py2,
-        but py3 slices of NodeList are returning a list
-        breaking slicing nodelist and refering to
-        properties and methods on contained object
-        """
-#        return self.__class__(self.data[index])
+    if sys.version_info < (3, 7):
+        # For PY3 before 3.7, slices of UserList/derivatives did not return
+        # the implementation class. See https://bugs.python.org/issue27639/.
+        # The solution there is simpler, but does not work for this class.
+        def __getitem__(self, index):
+            if isinstance(index, slice):
+                # Expand the slice object using range()
+                # limited by number of items in self.data
+                indices = index.indices(len(self.data))
+                return self.__class__([self[x] for x in range(*indices)])
 
-        if isinstance(index, slice):
-            # Expand the slice object using range()
-            # limited by number of items in self.data
-            indices = index.indices(len(self.data))
-            return self.__class__([self[x] for x in range(*indices)])
-
-        # Return one item of the tart
-        return self.data[index]
+            # Return one item of the tart
+            return self.data[index]
```

Comment:
no `__getitem__` needed for py >= 3.7 ?

Reply:
The inherited implementation from `UserList` works correctly, as noted in the referenced issue.